### PR TITLE
Futurize: render PKC compatible with Python 2 AND 3

### PR DESCRIPTION
**Pull request by davidheitman**
OK, after doing more reading, I started over with using futurize instead of 2to3.  This should hopefully ensure backward compatibility.  This is still not functional (websocket.py is very unhappy), but should be a better start.  I am very limited on time for now, so I might not be able to get back to this in the near future.

### CODE REVIEW

File: resources/lib/library_sync/full_sync.py [hunk 1]
```diff
@@ -121,8 +124,8 @@ def processing_loop_playstates(self):
             self.playstate_per_section(section)
 
     def playstate_per_section(self, section):
-        LOG.debug('Processing %s playstates for library section %s',
-                  section.number_of_items, section)
+#        LOG.debug('Processing %s playstates for library section %s',
+#                  section.number_of_items, section)
```
Comment (croneter) [inline resources/lib/library_sync/full_sync.py]: Codacy Issue found: [expected an indented block (comment)](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

**Comment by croneter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763&bid=16068271)

File: resources/lib/library_sync/fill_metadata_queue.py [hunk 2]
```diff
@@ -28,8 +30,8 @@ def __init__(self, repair, section_queue, get_metadata_queue):
     def _process_section(self, section):
         # Initialize only once to avoid loosing the last value before we're
         # breaking the for loop
-        LOG.debug('Process section %s with %s items',
-                  section, section.number_of_items)
+#        LOG.debug('Process section %s with %s items',
```
Comment (croneter) [inline resources/lib/library_sync/fill_metadata_queue.py]: Codacy Issue found: [expected an indented block (comment)](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/plexbmchelper/httppersist.py [hunk 3]
```diff
@@ -15,17 +19,17 @@
...
+class RequestMgr(object):
     def __init__(self):
         self.conns = {}
 
     def getConnection(self, protocol, host, port):
         conn = self.conns.get(protocol + host + str(port), False)
         if not conn:
             if protocol == "https":
-                conn = httplib.HTTPSConnection(host, port)
+                conn = http.client.HTTPSConnection(host, port)
```
Comment (croneter) [inline resources/lib/plexbmchelper/httppersist.py]: Codacy Issue found: [Use of HTTPSConnection on older versions of Python prior to 2.7.9 and 3.4.3 do not provide security, see https://wiki.openstack.org/wiki/OSSN/OSSN-0033](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/library_sync/fill_metadata_queue.py [hunk 4]
```diff
@@ -28,8 +30,8 @@ def __init__(self, repair, section_queue, get_metadata_queue):
...
+#                  section, section.number_of_items)
```
Comment (croneter) [inline resources/lib/library_sync/fill_metadata_queue.py]: Codacy Issue found: [expected an indented block (comment)](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/websocket.py [hunk 5]
```diff
@@ -1,917 +1,924 @@
...
+            if v != r:
+                return False
+
+        result = headers.get("sec-websocket-accept", None)
+        if not result:
+            return False
+        result = result.lower()
+
+        value = (key + '258EAFA5-E914-47DA-95CA-C5AB0DC85B11').encode()
+        hashed = base64.encodestring(hashlib.sha1(value).digest()).strip().lower().decode('utf-8')
```
Comment (croneter) [inline resources/lib/websocket.py]: Codacy Issue found: [Use of insecure MD2, MD4, MD5, or SHA1 hash function.](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/windows/kodigui.py [hunk 6]
```diff
@@ -281,7 +285,7 @@ def _updateListItem(self):
         self.listItem.setIconImage(self.iconImage)
         self.listItem.setThumbnailImage(self.thumbnailImage)
         self.listItem.setPath(self.path)
-        for k in self._manager._properties.keys():
+        for k in list(self._manager._properties.keys()):
```
Comment (croneter) [inline resources/lib/windows/kodigui.py]: Codacy Issue found: [Access to a protected member _properties of a client class](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/utils.py [hunk 7]
```diff
@@ -4,13 +4,17 @@
...
+from builtins import object
 from logging import getLogger
 from sqlite3 import OperationalError
 from datetime import datetime
 from unicodedata import normalize
 from threading import Lock
-import urllib
-import urlparse as _urlparse
+import urllib.request, urllib.parse, urllib.error
+import urllib.parse as _urlparse
```
Comment (croneter) [inline resources/lib/utils.py]: Codacy Issue found: [Reimport 'urllib.parse' (imported line 12)](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/library_sync/full_sync.py [hunk 8]
```diff
@@ -121,8 +124,8 @@ def processing_loop_playstates(self):
             self.playstate_per_section(section)
 
     def playstate_per_section(self, section):
-        LOG.debug('Processing %s playstates for library section %s',
-                  section.number_of_items, section)
+#        LOG.debug('Processing %s playstates for library section %s',
```
Comment (croneter) [inline resources/lib/library_sync/full_sync.py]: Codacy Issue found: [expected an indented block (comment)](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

File: resources/lib/library_sync/nodes.py [hunk 9]
```diff
@@ -192,7 +195,7 @@
          {
               'mode': 'browseplex',
               'key': ('/library/sections/{self.section_id}/recentlyViewed&%s'
-                      % urllib.urlencode({'type': v.PLEX_TYPE_NUMBER_FROM_PLEX_TYPE[v.PLEX_TYPE_EPISODE]})),
+                      % urllib.parse.urlencode({'type': v.PLEX_TYPE_NUMBER_FROM_PLEX_TYPE[v.PLEX_TYPE_EPISODE]})),
```
Comment (croneter) [inline resources/lib/library_sync/nodes.py]: Codacy Issue found: [Line too long (114/100)](https://app.codacy.com/manual/croneter/PlexKodiConnect/pullRequest?prid=4838763)

**Comment by davidheitman** [comment]
Planning to start work on this again soon.  Hopefully have a working version in a week or two, time permitting.  Good project to work on when you're trapped at home!

**Comment by croneter** [comment]
Ouch, sorry to hear :disappointed:

**Comment by croneter** [comment]
Be sure to `git rebase -i beta-version` to avoid conflicts by basing everything on the latest pkc beta. Also, if you frequently push your work, I might be able to give you a hint of two, thus hopefully reducing and fixing later on :wink:

**Comment by aejsmith** [comment]
@davidheitman Did you make any progress with this? I've just upgraded my Kodi PC to Fedora 32 and discovered PKC no longer works because Kodi is now using Python 3.

If you haven't had time to do anything more with this I might take a look sometime next week.

**Comment by croneter** [comment]
Superseded by https://github.com/croneter/PlexKodiConnect/tree/python3

**Pull request closed without merging**